app/persistence/postgresql/PostgresqlRepository.scala

```diff
@@ -602,6 +609,30 @@ class PostgresqlRepository @Inject() (
 
     }
 
+    def SELECT_DISTINCT(from: String, query: SpatialQuery, projection: SimpleProjection) = {
+
+      val cond = condition(query)
+
+      // construct string ->'properties'->>'foo'
+      val path = {
+        projection.path.path.collect {
+          case KeyPathNode(key) => key
+        }.map(str => s"'$str'").reverse match {
+          case head :: tail => s"->>$head" :: tail.map(str => s"->$str")
+          case Nil => Nil
+        }
+      }.reverse.mkString
+
+      val select = "distinct json" + path
```

Comment:
This works only for JSON tables, not for "regular" tables.  Need to check metadata.jsonTable property (see other query methods)

Reply:
See additional commit.